client/src/components/ResultsContainer.js

```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
```

Comment:
Reorganized state and effects to better see all of the state at once. Also, removed some redundant state variables

Reply:
good.